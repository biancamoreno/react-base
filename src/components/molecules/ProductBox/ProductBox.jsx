import React from "react";
import PropTypes from "prop-types";

import "./ProductBox.scss";

 const ProductBox = (props) => {

    const {description, price, image, location} = props;

    return (
        <div className="product-box__container">
            <div className="product-box__image">
                <img src={image} alt="product image"/>
            </div>
            <div className="product-box__descriptions">
                <h2 className="product-box__price">{price}</h2>
                <p className="product-box__title">{description}</p>
            </div>
            <div className="product-box__location">{location}</div>
        </div>
    )
}

ProductBox.propTypes = {
    description: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
    location:PropTypes.string.isRequired
}

export default ProductBox;
