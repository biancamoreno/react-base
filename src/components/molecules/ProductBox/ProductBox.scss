@import '../../../assets/variables/variables.scss';


.product-box__container {
    background-color: white;
    width: 80%;
    margin: 0 auto;
}