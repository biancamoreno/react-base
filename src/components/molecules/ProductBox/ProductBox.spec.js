import React from 'react';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';
import ProductBox from './ProductBox';

describe('ProductBox test', () => {
    it('should render correctly with props', () => {
        const props = {
            description: 'Apple iPhone 8 Plus 256 Gb Gris Espacial',
            price: 129999,
            image: 'http://mla-s2-p.mlstatic.com/951232-MLA31003080341_062019-I.jpg',
            location: 'Capital Federal',
            id: 'MLA820795723'
        }

        const wrapper = shallow(<ProductBox {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

});