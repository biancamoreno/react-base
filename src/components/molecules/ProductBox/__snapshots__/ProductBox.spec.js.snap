// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBox test should render correctly with props 1`] = `
<Link
  className="product-box__container"
  key="MLA820795723"
  to="/MLA820795723"
>
  <div
    className="product-box__image-container"
  >
    <img
      alt="product image"
      className="product-box__image"
      src="http://mla-s2-p.mlstatic.com/951232-MLA31003080341_062019-I.jpg"
    />
  </div>
  <div
    className="product-box__descriptions"
  >
    <h2
      className="product-box__price"
    >
      129999
    </h2>
    <p
      className="product-box__title"
    >
      Apple iPhone 8 Plus 256 Gb Gris Espacial
    </p>
  </div>
  <div
    className="product-box__location"
  >
    Capital Federal
  </div>
</Link>
`;
